File: Src2/Icm.Core/Functions/InterpolatedValueLine.vb
Imports Icm.Collections.Generic.StructKeyStructValue

Namespace Icm.Functions

    ''' <summary>
    ''' Interpolated keyed function with Date key and Double value.
    ''' </summary>
    ''' <remarks></remarks>
    Public Class InterpolatedValueLine
        Inherits InterpolatedKeyedFunction(Of Date, Double)

        Public Sub New(ByVal initialValue As Double, ByVal coll As ISortedCollection(Of Date, Double))
            MyBase.New(initialValue, New DateTotalOrder, New DoubleTotalOrder, coll)
        End Sub

        Public Overrides Function EmptyClone() As IMathFunction(Of Date, Double)
            Return New InterpolatedValueLine(V0, KeyStore)
        End Function

    End Class

End Namespace

File: Src/Icm.Core/Functions/InterpolatedValueLine.cs
using Icm.Collections.Generic.StructKeyStructValue;

namespace Icm.Functions
{
	/// <summary>
	/// Interpolated keyed function with Date key and Double value.
	/// </summary>
	/// <remarks></remarks>
	public class InterpolatedValueLine : InterpolatedKeyedFunction<System.DateTime, double>
	{
		public InterpolatedValueLine(double initialValue, ISortedCollection<System.DateTime, double> coll) : base(initialValue, new DateTotalOrder(), new DoubleTotalOrder(), coll)
		{
		}

		public override IMathFunction<System.DateTime, double> EmptyClone()
		{
			return new InterpolatedValueLine(V0(), KeyStore);
		}
	}
}